broker-core/src/main/java/io/zeebe/broker/system/workflow/repository/processor/DeploymentDistributor.java

```diff
@@ -120,40 +121,54 @@ private void pushDeploymentToPartitions(long key) {
   private void deployOnMultiplePartitions(long key) {
     LOG.trace("Distribute deployment to other partitions.");
 
-    final Int2ObjectHashMap<NodeInfo> partitionLeaders = partitionListener.getPartitionLeaders();
-    if (partitionLeaders.size() - 1 == remainingPartitions.size()) {
-      distributeDeployment(key);
-    } else {
-      LOG.trace("Not enough leaders");
-      delayedRetry(key);
-    }
+    final PendingDeploymentDistribution pendingDeploymentDistribution =
+        pendingDistributions.get(key);
+    final DirectBuffer directBuffer = pendingDeploymentDistribution.getDeployment();
+    pendingDeploymentDistribution.setDistributionCount(partitionsToDistributeTo.size());
+
+    pushDeploymentRequest.reset();
+    pushDeploymentRequest.deployment(directBuffer).deploymentKey(key);
+
+    final IntArrayList modifyablePartitionsList = new IntArrayList();
+    partitionsToDistributeTo.forEachOrderedInt(value -> modifyablePartitionsList.add(value));
+
+    distributeDeployment(modifyablePartitionsList);
   }
 
-  private void delayedRetry(long key) {
+  private void distributeDeployment(IntArrayList partitionsToDistribute) {
+    final IntArrayList remainingPartitions =
+        distributeDeploymentToPartitions(partitionsToDistribute);
+
+    distributeDelayedToRemainingPartitions(remainingPartitions);
```

Comment:
The method is so small that you could merge it with `distributeDeployment()`. I think it is easier to read.